tests/conftest.py

```diff
@@ -39,6 +63,25 @@ def docker(event_loop):
 
 
 @pytest.fixture
+def swarm(event_loop, docker):
```

Comment:
The main reason I did it is for test isolation. I tend to always use "failfast" when running tests and when I realized that I couldn't run the test again because my daemon was still in Swarm mode, I realized that enter/leave should actually be done systematically to avoid being in an inconsistent state.

It makes all sense imho: a Docker in Swarm mode is something very different to handle than a simple Docker daemon:
1. you need to join/leave the swarm
2. you get access to the "services" API

So in my opinion it is so much clearer to have a fixture "swarm" for a Docker Swarm environment and "docker" for a simple Docker environment.

It provides also a few perks:
1. If you now run the tests on your own Docker instance, if the daemon is already in swarm mode, it will not work. I can imagine that people who run tests doesn't want the test to remove all their services.
2. If you are not in Swarm mode, it will give you back your Docker daemon in the state it was before running the tests because it leaves the Swarm no matter what.

So now why in `conftest.py` and not in `test_services.py`? Well, we could test other stuff of a Swarm than just the services.